lighthouse-core/test/report/html/renderer/category-renderer-test.js

```diff
@@ -338,6 +346,37 @@ describe('CategoryRenderer', () => {
       assert.equal(passedAudits.length, 0);
       assert.equal(failedAudits.length, 12);
     });
+
+    it('expands warning audit group', () => {
+      const category = sampleResults.reportCategories.find(c => c.id === 'pwa');
+      const categoryClone = JSON.parse(JSON.stringify(category));
+      categoryClone.auditRefs[0].result.warnings = ['Some warning'];
+
+      const auditDOM = renderer.render(categoryClone, sampleResults.categoryGroups);
+      const warningClumpEl = auditDOM.querySelector('.lh-clump--warning');
+      const isExpanded = warningClumpEl.hasAttribute('open');
+      assert.ok(isExpanded, 'Warning audit group should be expanded by default');
+    });
+
+    it('only passing audits with warnings show in warnings section', () => {
+      const shouldBeFailed = renderer._getClumpIdForAuditRef({
```

Comment:
we should avoid touching the internal implementation if we can. We could do something simple like
```js
it('only passing audits with warnings show in warnings section', () => {
  const failingWarning = 'Failed and warned';
  const passingWarning = 'A passing warning';
  const category = {
    id: 'test',
    title: 'Test',
    score: 0,

    auditRefs: [{
      id: 'failing',
      result: {
        id: 'failing',
        title: 'Failing with warning',
        description: '',
        scoreDisplayMode: 'numeric',
        score: 0,
        warnings: [failingWarning],
      },
    }, {
      id: 'passing',
      result: {
        id: 'passing',
        title: 'Passing with warning',
        description: '',
        scoreDisplayMode: 'numeric',
        score: 1,
        warnings: [passingWarning],
      },
    }],
  };
  const categoryDOM = renderer.render(category);

  const shouldBeFailed = categoryDOM.querySelectorAll('.lh-clump--failed .lh-audit');
  assert.strictEqual(shouldBeFailed.length, 1);
  assert.strictEqual(shouldBeFailed[0].id, 'failing');
  assert.ok(shouldBeFailed[0].textContent.includes(failingWarning));

  const shouldBeWarning = categoryDOM.querySelectorAll('.lh-clump--warning .lh-audit');
  assert.strictEqual(shouldBeWarning.length, 1);
  assert.strictEqual(shouldBeWarning[0].id, 'passing');
  assert.ok(shouldBeWarning[0].textContent.includes(passingWarning));
});
```
(or we could do like the other ones do and just modify an existing category)